qiita_pet/handlers/analysis_handlers.py

```diff
@@ -267,12 +263,12 @@ def post(self, analysis_id):
         else:
             rarefaction_depth = None
         analysis = Analysis(analysis_id)
-        check_analysis_access(User(user), analysis)
+        check_analysis_access(self.current_user, analysis)
 
         command_args = self.get_arguments("commands")
         split = [x.split("#") for x in command_args]
 
-        moi_user_id = get_id_from_user(user)
+        moi_user_id = get_id_from_user(str(self.current_user))
```

Comment:
Doesn't the user has the property `id`? I saw that you added the `__str__` method to the user object, but the `QiitaUser` class extends the `QiitaBase` clase which already has the `id` property define. I would say that this is duplicating information and we can potentially use `__str__` in the future to output some other more important information rather than just the id, and we already have the `id` property to retrieve that information.